include/nuttx/timers/oneshot.h

```diff
@@ -229,6 +229,10 @@ struct oneshot_lowerhalf_s
 
   FAR const struct oneshot_operations_s *ops;
 
+#ifndef CONFIG_SCHED_TICKLESS
+  clock_t current_tick;
```

Comment:
but not all oneshot user need this field.

Reply:
This field is non-TICKLESS private. similar global ticks should be bound to oneshot instances.

Reply:
g_current_tick is bound to g_oneshot_lower, so it doesn't work by just moving g_current_tick to oneshot, but ignoring g_oneshot_lower.

Reply:
we have the per-cpu change for g_oneshot_lower, Moving g_current_tick to g_oneshot_lower is reasonable since it is instance-bound to each CPU.